torch/backends/_nnapi/serializer.py

```diff
@@ -1664,11 +1676,54 @@ def add_conv2d_common(
             inputs[9] = self.add_immediate_int_scalar(fuse_code)
             inputs[10] = self.add_immediate_bool_scalar(use_nchw)
 
+        _, _, in_h, in_w = image_oper.shape
         outputs = [None] * 1
-        outputs[0] = self.add_tensor_operand(jit_out, out_oper)
+        out_shape = get_conv_pool_shape(image_oper.shape, args, out_c, transpose)
+        out_oper = image_oper._replace(
+            shape=out_shape,
+            scale=out_scale,
+            zero_point=out_zero_point,
+        )
+        out_id = self.add_tensor_operand(jit_out, out_oper)
+        self._handle_conv_pool_var_ip(image_id, in_h, in_w, args, transpose, out_id)
 
+        outputs[0] = out_id
         self.add_operation(opcode, inputs, outputs)
 
+    def _handle_conv_pool_var_ip(self, image_id, in_h, in_w, args, transpose, out_id):
```

Comment:
What is "var_ip"?

I suggest making out_id the first argument after self.  I just have the feeling that if we need to add more options to the end, it will look cleaner that way.